lib/net/http.rb

```diff
@@ -942,14 +942,16 @@ def connect
       end
 
       D "opening connection to #{conn_address}:#{conn_port}..."
-      s = Timeout.timeout(@open_timeout, Net::OpenTimeout) {
+      s =
         begin
-          TCPSocket.open(conn_address, conn_port, @local_host, @local_port)
+          Socket.tcp(conn_address, conn_port, @local_host, @local_port,
+                     connect_timeout: @open_timeout)
```

Comment:
Thank you for this PR!
With the patch from https://bugs.ruby-lang.org/issues/15553, the parameter should be `resolv_timeout: @open_timeout`.

Otherwise, I can confirm this patch working on Linux.

If you're looking for a way to write a test for this, something like this should work: https://gist.github.com/kirs/5f711099b23ddae7a87ebb082ce43f59